src/xdp-utils.c

```diff
@@ -189,6 +189,40 @@ _xdp_parse_app_id_from_unit_name (const char *unit)
 
   return g_steal_pointer (&app_id);
 }
+
+char *
+_xdp_parse_app_id_from_dbus_name (const char *unit)
+{
+  g_autoptr(GRegex) regex = NULL;
+  g_autoptr(GMatchInfo) match = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *app_id = NULL;
+
+  g_assert (g_str_has_prefix (unit, "dbus-"));
+
+  // Parses a unit name such as dbus-:1.2-org.kde.kdeconnect@7.service
```

Comment:
This naming convention for systemd units is specific to dbus-broker.

On systems that have systemd activation available, the reference implementation (dbus-daemon) does not guarantee that activated services are in any sort of systemd service or scope: D-Bus services with `SystemdService=foo.service` are run as `foo.service` (probably also true for dbus-broker), and D-Bus services without `SystemdService` are run via "traditional activation" as children of `dbus.service` (unlike dbus-broker).

Reply:
Having said that, if a unit name matches this pattern then you can probably be quite confident that it's named according to how dbus-broker does it.